File: leetcode/MajorityElements.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        cnt = 1
        res = nums[0]
        for i in range(1, n):
            if nums[i] == res:
                cnt += 1
            else:
                cnt -= 1
                if cnt < 0:
                    cnt = 1
                    res = nums[i]
        return res

File: leetcode/MajorityElements.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int el = nums[0], cnt = 1, size = nums.size(), i;
        
        for(i = 1; i < size; i++) {
            (nums[i] == el) ? cnt++ : cnt--;
            if(!cnt) {
                el = nums[i];
                cnt = 1;
            }            
        }
        
        return el;
    }
};
